src/planner/World.cpp

```diff
@@ -151,5 +152,52 @@ std::mutex& World::getMutex() const
   return mMutex;
 }
 
+//==============================================================================
+bool World::State::operator==(const State& other) const
+{
+  return configurations == other.configurations;
+}
+
+//==============================================================================
+bool World::State::operator!=(const State& other) const
+{
+  return !(*this == other);
+}
+
+//==============================================================================
+World::State World::getState() const
+{
+  World::State state;
+
+  for (auto skeleton : mSkeletons)
+    state.configurations[skeleton->getName()] = skeleton->getConfiguration();
+
+  return state;
+}
+
+//==============================================================================
+void World::setState(const World::State& state)
+{
+  if (state.configurations.size() != mSkeletons.size())
+    throw std::invalid_argument(
+        "World::State and this World do not have the same number of "
+        "skeletons.");
+
+  for (const auto& skeleton : mSkeletons)
+  {
+    auto name = skeleton->getName();
+    auto it = state.configurations.find(name);
+    if (it == state.configurations.end())
+      throw std::invalid_argument(
+          "Skeleton " + name + " does not exist in state.");
+  }
+
+  for (const auto& skeleton : mSkeletons)
+  {
+    std::lock_guard<std::mutex> lock(skeleton->getMutex());
+    skeleton->setConfiguration(state.configurations.at(skeleton->getName()));
```

Comment:
Can we use `state.configurations[skeleton->getName()]` instead of `at`? Since we've already checked that the keys should all exist, `at` seems unnecessary.

Reply:
Actually, maybe we could merge the two loops and then just use `at` instead of explicitly calling `find` and then indexing? @jslee02 is it inefficient to catch the (possible) `out_of_range` exception from `at` and then trigger our own more helpful exception?

Reply:
Hmm. Can't do `[]` because `[]` is a non-const operator and I'm taking in `const State&`. 
https://stackoverflow.com/questions/15660838/error-no-match-for-operator-in-near-match

I'll keep it as is.